ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Connection.scala

```diff
@@ -140,14 +141,21 @@ private[h2] class H2Connection[F[_]](
     } >>
       H2Connection.KillWithoutMessage().raiseError
 
+  private[this] def mapRefForEach(f: H2Stream[F] => F[Unit]): F[Unit] = {
```

Comment:
Isn't this just `traverse_`

Reply:
m.valuesIterator traverse_ seems like we could do somehow. That would be simpler then methods with iterator.

Reply:
Ah, well... the problem is that there is no `traverse_` for `Iterator`, there is only a `Foldable` for `Iterable`, and then implementation of `traverse_` of `cats` also incurs memory cost. 

```Scala
  def traverse_[G[_], A, B](fa: F[A])(f: A => G[B])(implicit G: Applicative[G]): G[Unit] =
    foldRight(fa, Always(G.pure(()))) { (a, acc) =>
      G.map2Eval(f(a), acc) { (_, _) =>
        ()
      }
    }.value
```